packages/apple-pay-integration/src/apple-pay-payment-strategy.ts

```diff
@@ -209,9 +219,17 @@ export default class ApplePayPaymentStrategy implements PaymentStrategy {
         promise: ApplePayPromise,
     ) {
         const { token } = event.payment;
+
+        let deviceSessionId: string | undefined;
+
+        if (paymentMethod.initializationData?.gateway === ApplePayGatewayType.BRAINTREE) {
+            deviceSessionId = await this._getDeviceData();
```

Comment:
Perhaps we can instantiate a `braintreeIntegrationService` here and get device data.

Reply:
Yes, by doing so, we can remove `private _braintreeIntegrationService: BraintreeIntegrationService` from the constructor.